fem/eltrans.cpp

```diff
@@ -226,29 +267,24 @@ int IsoparametricTransformation::TransformBack(const Vector &pt,
       // or when dim != sdim: x := x + [J^t.J]^{-1}.J^t [pt-F(x)]
       Transform(xip, y);
       subtract(pt, y, y); // y = pt-y
-      if (y.Normlinf() < phys_tol) { ip = xip; return 0; }
+      if (y.Normlinf() < phys_tol)
+      {
+         ip = xip;
+         return Geometry::CheckPoint(geom, ip, ip_tol ) ? 0 : 1;
+      }
+
       SetIntPoint(&xip);
       CalcInverse(Jacobian(), Jinv);
       Jinv.Mult(y, dx);
       x += dx;
-      prev_xip = xip;
-      prev_hit_bdr = hit_bdr;
       xip.Set(xd, dim); // x -> xip
-      // If xip is ouside project it on the boundary on the line segment
```

Comment:
That seems like a reasonable thing to do... interesting that it leads to bad behavior.